### DISCUSSION TITLE
[A-L] (22 maggio 2024) Esercizio 3

### DISCUSSION THREAD

**Discussion by Elia-Belli** [sapienzastudentsnetwork/progettazione-di-algoritmi#102] [Esercitazioni A-L]
![immagine](https://github.com/user-attachments/assets/16e97df7-30df-4dd8-aca5-7e13386a6531)

**Comment by daromacarlo** [comment]
Per ogni nodo eseguo una DFS per vedere quali altri nodi può raggiungere.
![immagine](https://github.com/user-attachments/assets/89526416-3852-41f3-aeb5-c5edd5abbb54)

eseguo una semplice DFS:
![immagine](https://github.com/user-attachments/assets/3b7a878b-daec-4e7d-a5f2-66385ca242b1)

T(n) = O((n+m) * n) = O(n^2 + n * m) 
O(n * m) = O(n * n * (n-1)) nel caso peggiore (quando ogni nodo è collegato a ogni altro nodo), quindi
T(n) = O(n * m)
